pkg/virt-controller/watch/vmi/vmi_test.go

```diff
@@ -3609,6 +3609,81 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			expectMatchingPodCreation(vmi)
 		})
 	})
+
+	Context("Event handling", func() {
+		var (
+			vmi *virtv1.VirtualMachineInstance
+		)
+
+		indexVirtualMachine := func(vmi *virtv1.VirtualMachineInstance) {
+			Expect(controller.vmiIndexer.Add(vmi)).To(Succeed())
+			_, err := virtClientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Create(context.Background(), vmi, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		pvcWith := func(requestSize, capacity, rv string) *k8sv1.PersistentVolumeClaim {
+			parsedReq, _ := resource.ParseQuantity(requestSize)
+			parsedCap, _ := resource.ParseQuantity(capacity)
+			pvc := newPvc("default", "testpvc")
+			pvc.Spec.Resources.Requests = k8sv1.ResourceList{
+				k8sv1.ResourceStorage: parsedReq,
+			}
+			pvc.Status.Capacity = k8sv1.ResourceList{
+				k8sv1.ResourceStorage: parsedCap,
+			}
+			pvc.ObjectMeta.ResourceVersion = rv
+			return pvc
+		}
+
+		BeforeEach(func() {
+			vmi = &virtv1.VirtualMachineInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "testvmi",
+					Namespace: "default",
+				},
+				Spec: virtv1.VirtualMachineInstanceSpec{
+					Volumes: []virtv1.Volume{
+						{
+							Name: "myvolume",
+							VolumeSource: virtv1.VolumeSource{
+								DataVolume: &virtv1.DataVolumeSource{
+									Name: "testpvc",
+								},
+							},
+						},
+					},
+				},
+			}
+			indexVirtualMachine(vmi)
+		})
+
+		DescribeTable("updatePVC", func(oldReq, oldCap, newReq, newCap, oldRV, newRV string, expectEnqueue bool) {
+			oldPVC := pvcWith(oldReq, oldCap, oldRV)
+			newPVC := pvcWith(newReq, newCap, newRV)
+
+			addDataVolumePVC(oldPVC)
+			controller.updatePVC(oldPVC, newPVC)
+
+			if expectEnqueue {
+				Eventually(func() int {
+					return mockQueue.Len()
+				}, time.Second, 100*time.Millisecond).Should(Equal(1))
+
+				item, _ := mockQueue.Get()
+				Expect(item).To(Equal("default/testvmi"))
+			} else {
+				Consistently(func() int {
+					return mockQueue.Len()
+				}, time.Second).Should(Equal(0))
```

Comment:
Use same interval as above? This will only query once IIUC?

Reply:
ack